pkg/server/server.go

```diff
@@ -97,20 +97,42 @@ func registerControllers(API *web.API, router *mux.Router, config *Settings) {
 		for _, route := range ctrl.Routes() {
 			log.D().Debugf("Registering endpoint: %s %s", route.Endpoint.Method, route.Endpoint.Path)
 			handler := web.Filters(API.Filters).ChainMatching(route)
-			router.Handle(route.Endpoint.Path, api.NewHTTPHandler(handler, config.MaxBodyBytes)).Methods(route.Endpoint.Method)
+			apiHandler := api.NewHTTPHandler(handler, config.MaxBodyBytes)
+			if !route.DisableHTTPTimeouts {
+				router.Handle(route.Endpoint.Path, newContentTypeHandler(http.TimeoutHandler(apiHandler, config.RequestTimeout, `{"error":"Timeout", "description": "operation has timed out"}`))).Methods(route.Endpoint.Method)
```

Comment:
should this timeout be min(config.RequestTimeout, servertimeout-1sec) ? Or we should have config validation that servertimeout is bigger than request timeout

Reply:
May be validation of the configuration will be best

Reply:
actually, if you check the creation of the Server struct, I have already put there for WriteTimeout +1, this will make sure that we have time to write the response before completely closing the response.